tests/context_tests.rs

```diff
@@ -1371,3 +1371,76 @@ mod test_hash {
         assert_ne!(ticket.digest().len(), 0); // Should do some better checking of the digest
     }
 }
+
+mod test_policy_get_digest {
+    use super::*;
+    #[test]
+    fn get_policy_digest() {
+        let mut context = create_ctx_without_session();
+        let trial_session = context
+            .start_auth_session(
+                ESYS_TR_NONE,
+                ESYS_TR_NONE,
+                &[],
+                TPM2_SE_TRIAL,
+                utils::TpmtSymDefBuilder::aes_256_cfb(),
+                TPM2_ALG_SHA256,
+            )
+            .unwrap();
+        let trial_session_attr = TpmaSessionBuilder::new()
+            .with_flag(TPMA_SESSION_DECRYPT)
+            .with_flag(TPMA_SESSION_ENCRYPT)
+            .build();
+        context
+            .tr_sess_set_attributes(trial_session, trial_session_attr)
+            .unwrap();
+
+        // Read the pcr values using pcr_read
+        let pcr_selections = PcrSelectionsBuilder::new()
+            .with_selection(HashingAlgorithm::Sha256, &[PcrSlot::Slot0, PcrSlot::Slot1])
+            .build();
+
+        let (_update_counter, pcr_selections, pcr_data) = context.pcr_read(pcr_selections).unwrap();
+
+        // Run pcr_policy command.
+        //
+        // "If this command is used for a trial policySession,
+        // policySession→policyDigest will be updated using the
+        // values from the command rather than the values from a digest of the TPM PCR."
+        //
+        // "TPM2_Quote() and TPM2_PolicyPCR() digest the concatenation of PCR."
+        let concatenated_pcr_values = [
+            pcr_data
+                .pcr_bank(HashingAlgorithm::Sha256)
+                .unwrap()
+                .pcr_value(PcrSlot::Slot0)
+                .unwrap()
+                .value(),
+            pcr_data
+                .pcr_bank(HashingAlgorithm::Sha256)
+                .unwrap()
+                .pcr_value(PcrSlot::Slot1)
+                .unwrap()
+                .value(),
+        ]
+        .concat();
+
+        let (hashed_data, _ticket) = context
+            .hash(
+                &concatenated_pcr_values,
+                HashingAlgorithm::Sha256,
+                Hierarchy::Owner,
+            )
+            .unwrap();
+        // There should be no errors setting pcr policy for trial session.
+        context
+            .policy_pcr(trial_session, &hashed_data, pcr_selections)
+            .unwrap();
+
+        // There is now a policy digest that can be retrived and used.
+        let retrieved_policy_digest = context.policy_get_digest(trial_session).unwrap();
```

Comment:
I'm assuming the next step would be to use this value to control access to something? (legit curious, never used PCR policy sessions)

Reply:
Exactly!
It can be used for authorizing access instead or together with a password.